Lib/test/test_pdb.py

```diff
@@ -3235,6 +3237,32 @@ def test_checkline_is_not_executable(self):
                 self.assertFalse(db.checkline(os_helper.TESTFN, lineno))
 
 
+@support.requires_subprocess()
+class PdbTestReadline(unittest.TestCase):
+    def setUpClass():
+        # Ensure that the readline module is loaded
+        # If this fails, the test is skipped because SkipTest will be raised
+        readline = import_module('readline')
+        if readline.__doc__ and "libedit" in readline.__doc__:
+            raise unittest.SkipTest("libedit readline is not supported for pdb")
+
+    def test_basic_completion(self):
+        script = textwrap.dedent("""
+            import pdb; pdb.Pdb().set_trace()
+            print('hello')
+        """)
+
+        # List everything starting with 'co', there should be multiple matches
+        # then add ntin and complete 'contin' to 'continue'
+        input = b"co\t\tntin\t\n"
+
+        output = run_pty(script, input)
+
+        self.assertIn(b'cont', output)
+        self.assertIn(b'condition', output)
+        self.assertIn(b'continue', output)
```

Comment:
The `cont` assert is useless since `continue` is checked later. How about checking that it appears as a separate word?

Also, IMO it would be nice to assert that the`continue` command caused the `print` to run (outputting a string that doesn't appear in the source).
```suggestion
        self.assertIn(b'cont', output.split())
        self.assertIn(b'condition', output)
        self.assertIn(b'continue', output)
        self.assertIn(b'hello!', output)
```